packages/blade/src/components/Button/BaseButton/BaseButton.tsx

```diff
@@ -49,32 +51,43 @@ const StyledBaseText = styled(BaseText)(
 );
 
 type BaseButtonStyleProps = {
-  iconSize: IconSize; // prop
-  fontSize: keyof Theme['typography']['fonts']['size']; // prop
-  lineHeight: keyof Theme['typography']['lineHeights']; // prop
-  minHeight: '48px' | '40px' | '32px' | '28px'; // raw value
-  iconSpacing: string; // raw value
-  iconColor: IconProps['color']; // prop
-  textColor: BaseTextProps['color']; // prop
-  spacing: string; // raw value
+  iconSize: IconSize;
+  fontSize: keyof Theme['typography']['fonts']['size'];
+  lineHeight: keyof Theme['typography']['lineHeights'];
+  minHeight: ButtonMinHeight;
+  iconSpacing: string;
+  iconColor: IconProps['color'];
+  textColor: BaseTextProps['color'];
+  spacing: string;
   text?: string;
+  color: string;
+  borderColor: string;
+  hoverColor: string;
+  hoverBorderColor: string;
+  activeColor: string;
+  activeBorderColor: string;
+  focusColor: string;
+  focusBorderColor: string;
+  focusRingColor: string;
```

Comment:
What is the difference between `focusBorderColor` and `focusRingColor`? 🤔

Reply:
Why do we need both?

Reply:
Is the `box-shadow` used in focus different from the one that might be used in hover (if we decide to keep one)? If not, then we can consolidate it in a single var.  Also, does it make sense to have this one called `focusShadowColor` instead of `focusRingColor` for more clarity?

Reply:
>Why do we need both?

@anuraghazra Border always exists even on default state. We can't treat border as a focus ring. See the tertiary variant design for example where you have border along with a focus ring

Reply:
@nashcheez For now we're not keeping box-shadow on hover. We would have to remove the hover box-shadow on focus to show the focus ring instead. Could be a problem maybe if design wants shadow along with focus ring. But we can discuss that with design separately. 

Right now, the color is acting as a `focusRing` so lets call it that. If we do want shadows later on we can rename this. But as of now, it's better to keep the naming as per its usage.